wp/sections/compare.tex

```diff
@@ -0,0 +1,48 @@
+\par A lot of projects uses ktlint as their code formatting tool. For example, OmiseGo \footnote{\url{https://github.com/omgnetwork/android-sdk}} (currently rebranding to OMG Network) - a quite popular cryptocurrency.
+
+\subsection{About detekt}
+\par Detekt is a static code analysis tool. It operates on an abstract syntax tree provided by Kotlin compiler. As it was mentioned in ktlint section, it has it's own rules. Detekt uses ktlint to redefine it's formatting rules. Detekt supports such features as code smell analysis, highly configurable rule sets, IntelliJ integration and third-party integrations for Maven, Bazel and Github actions and many more. It has been developing since 2016 and today it has 3.2k stars, 411 forks and 1850 closed PRs. It has around 45k lines of code.
+\par Detekt is used in such projects as fountain or Kaspresso. Fountain "is an Android Kotlin library conceived to make your life easier when dealing with paged endpoint services" \footnote{\url{https://github.com/xmartlabs/fountain}} and Kaspresso is a framework for UI testing on Android made by KasperskyLab \footnote{\url{https://github.com/KasperskyLab/Kaspresso}}.
+
+\subsection{About ktfmt}
+\par Ktfmt is a program that formats Kotlin code, based on google-java-format. It's development has started in Facebook in the end of 2019. It can be added to your project through a Maven dependency, Gradle dependency, IntelliJ plugin or you can run it through a command line. Ktfmt is not a configurable, so to change any rule logic you need to download the project and redefine some constants. Ktfmt has 214 stars, 16 forks, 20 closed PRs and around 7500 lines of code. 
+
+\subsection{About diKTat}
+\par Diktat as well as ktlint and detekt is a static code analysis tool. It's development started in 2020 and at the time of writing this whitepaper diKTat has 130 stars and 12 forks. DiKTat operates on AST provided by IntelliJ. 
+\par So why is diKTat better? First of all, we support much more rules than ktlint. Our ruleset includes more than 100 rules. Second, diKTat is configurable. A lot of rules have it's own settings, which can be easily understood. For example, you can choose whether you need a copyright or choose a length of line. Third, diKTat is very easy to configure. You don't need to spend hours only to understand what each rule is doing. Our ruleset is a yml file, where each rule is commented out. Last but not the least, diKTat can be used as a CI/CD tool in order to avoid merging errors in the code.
```

Comment:
```suggestion
\par So why is diKTat better? First of all, we support much more rules than ktlint. Our ruleset includes more than 100 rules. Second, diKTat is configurable. A lot of rules have their own settings, and all of them can be easily understood. For example, you can choose whether you need a copyright or choose a length of line. Third, diKTat is very easy to configure. You don't need to spend hours only to understand what each rule is doing. Our ruleset is a yml file, where each rule is commented out. Last but not the least, diKTat can be used as a CI/CD tool in order to avoid merging errors in the code.
```